velox/connectors/hive/HiveDataSink.h

```diff
@@ -378,13 +378,22 @@ struct HiveWriterId {
 
   HiveWriterId() = default;
 
+  /// HiveWriterId for partitioned (but non-bucketed tables).
```

Comment:
This becomes confusing now. How about we provide a set of static create methods? 
```
static createForPartitionedTable,
static createForUnPartitionTable
static createForPartitionAndBucketedTable
static createForBucketedTable?
```
And have a private method which take both parameter as optionals? thanks!

Reply:
@xiaoxmeng : I initially stayed away from changing the initial constructors but then realized they could all be subsumed by a single one

  HiveWriterId(
      std::optional<uint32_t> _partitionId,
      std::optional<uint32_t> _bucketId = std::nullopt)

So have now left only this single one.